modules/WebViewBridge.ts

```diff
@@ -0,0 +1,36 @@
+import WebView from 'react-native-webview';
+
+export interface MemochatNativeMessage {
+  action: 'back';
+  data?: Record<string, unknown>;
+  callbackId?: string;
+}
+
+/**
+ * webview로 메시지 보내는 브릿지
+ */
+class WebViewBridge {
+  static instance: WebViewBridge;
+  static webViewRef: WebView;
+
+  constructor(webViewRef?: WebView) {
+    if (!WebViewBridge.instance) {
+      if (!webViewRef) {
+        throw new Error('WebView ref is not set');
+      }
+      WebViewBridge.webViewRef = webViewRef;
+      WebViewBridge.instance = this;
+      return WebViewBridge.instance;
+    }
+    return WebViewBridge.instance;
+  }
+
+  back() {
+    const msg: MemochatNativeMessage = {
+      action: 'back',
+    };
+    WebViewBridge.webViewRef.postMessage(JSON.stringify(msg));
+  }
+}
+
+export default WebViewBridge;
```

Comment:
WebviewBridge와 webviewvMessageReceiver가 싱글톤 패턴으로 하나의 객체만 생성되어야 하기 때문에, 이렇게 객체를 만들어서 export 하는 것이 어떨까용??
참고: https://www.patterns.dev/posts/singleton-pattern/
```suggestion

const webviewBridge = Object.freeze(new WebviewBridge());
export default webviewBridge;
```

ref는 setRef와 같은 함수로 설정해주는 것은 어떨지 싶습니다!

Reply:
ref가 없으면 사용할수 없는 객체이기 때문에 생성자에 명시적으로 ref를 넣어주는게 좋다고 생각했습니다
<img width="468" alt="image" src="https://user-images.githubusercontent.com/67774824/202891567-d197dfe9-f23d-4cca-9e4b-296777f276c5.png">
생성자에 방어코드를 넣어서 new 키워드로 객체를 생성해도 어디서든 싱글톤으로 유지될것 같은데 이대로 쓰는건 어떠실까요?